exec/testing/fake_exec.go

```diff
@@ -28,13 +33,34 @@ type FakeExec struct {
 	CommandScript []FakeCommandAction
 	CommandCalls  int
 	LookPathFunc  func(string) (string, error)
+
+	T *testing.T
+}
+
+// NewFakeExec creates a FakeExec. You can pass nil for t if you are not running from a
+// test program. (This will cause certain failures to result in a panic rather than a
+// t.Fatal.) cmdsInPath is a list of command names for which LookPath will return a match
+// ("/fake-bin/commandname"); if your code does not use LookPath you can pass nil. (If you
+// need more complex behavior, you can override LookPathFunc on the returned FakeExec.)
+func NewFakeExec(t *testing.T, cmdsInPath []string) *FakeExec {
+	return &FakeExec{
+		T: t,
+		LookPathFunc: func(cmd string) (string, error) {
+			for _, pathCmd := range cmdsInPath {
+				if pathCmd == cmd {
+					return "/fake-bin/" + cmd, nil
+				}
+			}
+			return "", &osexec.Error{Name: cmd, Err: osexec.ErrNotFound}
+		},
+	}
 }
 
 type FakeCommandAction func(cmd string, args ...string) exec.Cmd
 
 func (fake *FakeExec) Command(cmd string, args ...string) exec.Cmd {
 	if fake.CommandCalls > len(fake.CommandScript)-1 {
-		panic(fmt.Sprintf("ran out of Command() actions. Could not handle command [%d]: %s args: %v", fake.CommandCalls, cmd, args))
+		fake.fail("ran out of Command() actions at %s executing %v", getCaller(1), append([]string{cmd}, args...))
```

Comment:
Well, but the Command still has to do *something* when the code being tested calls `CombinedOutput()`/`Run()` on it. So you could have it return an error to the caller, but then you're testing how your code responds when it gets an error that doesn't correspond to anything it could actually get in the real world.

If you hit this point (or any of the other panic/fail points), then there is a bug in your test case, not in the code that it's testing. So the only reasonable behavior is to abort the test case.